local.tf

```diff
@@ -0,0 +1,566 @@
+##########################################
+
+resource "aws_s3_bucket" "file_storage" {
+  bucket        = "${var.global_environment_name}-wandb-files"
+  acl           = "private"
+  force_destroy = true
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["GET", "HEAD"]
+    allowed_origins = ["*"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+  }
+
+  server_side_encryption_configuration {
+    rule {
+        apply_server_side_encryption_by_default {
+          sse_algorithm = "AES256"
+        }
+    }
+  }
+}
+
+resource "aws_s3_bucket_notification" "file_metadata_sns" {
+  bucket = aws_s3_bucket.file_storage.id
+
+  topic {
+    topic_arn = aws_sns_topic.file_metadata.arn
+    events    = ["s3:ObjectCreated:*"]
+  }
+}
+
+##########################################
+# RDS
+##########################################
+
+resource "aws_db_subnet_group" "metadata_subnets" {
+  name       = "wandb-db-subnets"
+  subnet_ids = aws_subnet.wandb.*.id
+}
+
+resource "aws_rds_cluster" "metadata_cluster" {
+  engine               = "aurora-mysql"
+  db_subnet_group_name = aws_db_subnet_group.metadata_subnets.name
+
+  skip_final_snapshot     = true
+  backup_retention_period = 14
+
+  enabled_cloudwatch_logs_exports = [
+    "error",
+  ]
+  iam_database_authentication_enabled = true
+
+  database_name   = "wandb_local"
+  master_username = "wandb"
+  master_password = "wandb_root_password"
+
+  vpc_security_group_ids = [aws_security_group.metadata_store.id]
+
+  storage_encrypted = true
+}
+
+resource "aws_rds_cluster_instance" "metadata_store" {
+  identifier           = "wandb-metadata"
+  engine               = "aurora-mysql"
+  cluster_identifier   = aws_rds_cluster.metadata_cluster.id
+  instance_class       = "db.r5.large"
+  db_subnet_group_name = aws_db_subnet_group.metadata_subnets.name
+}
+
+resource "aws_security_group" "metadata_store" {
+  name        = "wandb-metadata-store"
+  description = "Allow inbound traffic from workers to metadata store"
+  vpc_id      = aws_vpc.wandb.id
+
+  tags = {
+    Name = "wandb-metadata-store"
+  }
+}
+
+resource "aws_security_group_rule" "metadata_ingress_eks_workers" {
+  description              = "Allow inbound traffic from EKS workers to metadata store"
+  from_port                = 3306
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.metadata_store.id
+  source_security_group_id = aws_eks_cluster.wandb.vpc_config[0].cluster_security_group_id
+  to_port                  = 3306
+  type                     = "ingress"
+}
+
+##########################################
+# Output files
+##########################################
+
+resource "local_file" "kubeconfig" {
+  filename = "kubeconfig.yaml"
+  content = <<KUBECONFIG
+    apiVersion: v1
+    kind: Config
+    clusters:
+    - cluster:
+        server: ${aws_eks_cluster.wandb.endpoint}
+        certificate-authority-data: ${aws_eks_cluster.wandb.certificate_authority[0].data}
```

Comment:
Does this mean AWS provisions a cert?

Reply:
Not quite? This is for cluster administration. Here are the terraform docs on the matter: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster#attributes-reference

I mean, conceptually, maybe, but this almost certainly isn't something you could use to set up HTTPS ingress to the cluster.